gobblin-modules/gobblin-kafka-09/src/main/java/gobblin/source/extractor/extract/kafka/KafkaSimpleStreamingExtractor.java

```diff
@@ -0,0 +1,216 @@
+import gobblin.source.extractor.extract.kafka.SimpleKafkaSchemaRegistry;
+import gobblin.writer.WatermarkStorage;
+
+
+/**
+ * An implementation of {@link StreamingExtractor} from which reads from Kafka and returns records as an array of bytes.
+ *
+ * @author Shrikanth Shankar
+ *
+ *
+ */
+public class KafkaSimpleStreamingExtractor extends EventBasedExtractor<String, RecordEnvelope<byte[]>> implements StreamingExtractor<String, byte[]>, WatermarkStorage {
+
+  public static class KafkaWatermark implements CheckpointableWatermark {
+    TopicPartition _topicPartition;
+    LongWatermark _lwm;
+
+    @VisibleForTesting
+    public KafkaWatermark(TopicPartition topicPartition, LongWatermark lwm) {
+      _topicPartition = topicPartition;
+      _lwm = lwm;
+    }
+
+    @Override
+    public String getSource() {
+      return _topicPartition.toString();
+    }
+
+    @Override
+    public ComparableWatermark getWatermark() {
+      return _lwm;
+    }
+
+    @Override
+    public short calculatePercentCompletion(Watermark lowWatermark, Watermark highWatermark) {
+      return 0;
+    }
+
+    @Override
+    public JsonElement toJson() {
+      return WatermarkSerializerHelper.convertWatermarkToJson(this);
+    }
+
+    @Override
+    public int compareTo(CheckpointableWatermark o) {
+      Preconditions.checkArgument(o instanceof KafkaWatermark);
+      KafkaWatermark ko = (KafkaWatermark)o;
+      Preconditions.checkArgument(_topicPartition.equals(ko._topicPartition));
+      return _lwm.compareTo(ko._lwm);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+      if (obj == null)
+        return false;
+      if (!(obj instanceof KafkaWatermark))
+          return false;
+      return this.compareTo((CheckpointableWatermark)obj) == 0;
+    }
+
+    @Override
+    public int hashCode() {
+      final int prime = 31;
+      return _topicPartition.hashCode() * prime + _lwm.hashCode();
+    }
+
+    public TopicPartition getTopicPartition() {return _topicPartition;};
+
+    public LongWatermark getLwm() {return  _lwm;};
+  }
+
+
+  private final KafkaSchemaRegistry<String, String> _kafkaSchemaRegistry;
+  private Consumer<String, byte[]> _consumer;
+  private TopicPartition _partition;
+  private Iterator<ConsumerRecord<String, byte[]>> _records;
+  long _rowCount = 0;
+
+  public KafkaSimpleStreamingExtractor(WorkUnitState state) {
+    super(state);
+    _kafkaSchemaRegistry = new SimpleKafkaSchemaRegistry(state.getProperties());
+    _consumer = KafkaSimpleStreamingSource.getKafkaConsumer(state);
+    closer.register(_consumer);
+    _partition = new TopicPartition(state.getProp(KafkaSimpleStreamingSource.TOPIC_NAME),
+                                    state.getPropAsInt(KafkaSimpleStreamingSource.PARTITION_ID));
+    _consumer.assign(Collections.singletonList(_partition));
+    OffsetAndMetadata offset = _consumer.committed(_partition);
+    if (offset == null)
+      _consumer.seekToBeginning(_partition);
```

Comment:
The regular kafka consumer allows to bootstrap from the beginning or from the end since the former may be time-consuming.